openedx/core/djangoapps/credentials_service/mixin.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import unicode_literals
+import os
+
+from django.conf import settings
+
+
+def assets_path(instance, filename):
```

Comment:
A mixin is not a file. Taking a closer look at how this is used, it can live in models.py, but place it near the top of the file instead of right in the middle of a bunch of class definitions. Take a look at the [documentation for upload_to](https://docs.djangoproject.com/en/1.9/ref/models/fields/#django.db.models.FileField.upload_to) again. This function is doing far more than it needs to do. You only need to return a path. `MEDIA_ROOT` should not be referenced at all.

I would also split this into two methods, one for each data type.

```python
def signatory_upload_to(instance, filename):
  return 'signatories/{id}/{filename}'.format(id=instance.id, filename=filename)

def certificate_template_asset_upload_to(instance, filename):
  return 'certificate_template_assets/{id}/{filename}'.format(id=instance.id, filename=filename)
```